docs/source/Plugin/P146.rst

```diff
@@ -0,0 +1,118 @@
+.. include:: ../Plugin/_plugin_substitutions_p14x.repl
+.. _P146_page:
+
+|P146_typename|
+==================================================
+
+|P146_shortinfo|
+
+Plugin details
+--------------
+
+Type: |P146_type|
+
+Name: |P146_name|
+
+Status: |P146_status|
+
+GitHub: |P146_github|_
+
+Maintainer: |P146_maintainer|
+
+Used libraries: |P146_usedlibraries|
+
+.. Supported hardware
+.. ------------------
+
+|P146_usedby|
+
+Description
+------------
+
+This plugin aids in (re)sending data, accumulated by the Cache Controller on the local file system, to a controller.
+
+It will re-send the data using the original plugin via the original controller it was supposed to be sent out, but no network connection was available, like when collecting data in a car, and sending it out once the home location, with WiFi is available again.
+
+Data will be sent out to the configured Controller, while the task(s) collecting data should be sending their data to the Cache Controller, that will have to be configured (and enabled!) separately on the Controllers tab.
+
+Device Configuration
+--------------------
+
+.. image:: P146_DeviceConfiguration.png
+
+
+* **Name**: Required by ESPEasy, must be unique among the list of available devices/tasks.
+
+* **Enabled**: The device can be disabled or enabled. When not enabled the device should not use any resources.
+
+* **Delete Cache Files After Send**: After successfully sending out the data, the cache files will be removed when this setting is enable. When leaving this option disabled, might cause data to be sent multiple time, depending on the receiving controller to deduplicate the received data.
+
+MQTT Output Options
+^^^^^^^^^^^^^^^^^^^
+
+* **Send buld**: When enabled will send the data in a bulk message when possible, when disabled separate messages will be sent.
+
+* **HEX Encoded Binary**: Determines if any binary data will be hex encoded when sent. Depends on the receiving (MQTT) controller is this option is needed.
+
+* **Minimal Send Interval**: The minimal required send interval, defaults to 100 msec, but when using an external MQTT server, this might need to be increased to accommodate the TOS (Terms of service).
+
+* **Max message size**: This is the max size for the content sent in a single message.
```

Comment:
This can be quite a large MQTT message, upto 32kB in size.
Practical limit depends on a lot of factors, like the MQTT broker configuration.

N.B. also CSV export can be sent in bulk, even with combining data based on the same timestamp.